src/transformers/integrations/integration_utils.py

```diff
@@ -751,6 +754,10 @@ def setup(self, args, state, model, **kwargs):
                 self._wandb.watch(model, log=_watch_model, log_freq=max(100, state.logging_steps))
             self._wandb.run._label(code="transformers_trainer")
 
+            # add number of model parameters to wandb config
+            if isinstance(model, (PreTrainedModel, TFPreTrainedModel)):
+                self._wandb.config["model/num_parameters"] = model.num_parameters()
```

Comment:
Does this have to be logged both on `setup` and `on_train_begin` and 'on_save`?

Reply:
In `setup` we log it to the config, in `on_train_begin` and `on_save` we log it to the model artifact metadata and the checkpoint metadata respectively. This way, the run config, and metadata for the model and checkpoint metadata will all have the model parameter making it easier for reproducibility. That was my thought process here.

Reply:
makes sense, thanks!